chips/cc26x0/src/uart.rs

```diff
@@ -168,26 +239,111 @@ impl UART {
         regs.icr.write(Interrupts::ALL_INTERRUPTS::SET);
     }
 
+    pub fn enable_interrupts(&self) {
+        let regs = unsafe { &*self.regs };
+        // Clear all UART interrupts
+        regs.icr.write(Interrupts::ALL_INTERRUPTS::SET);
+        // Enable all UART interrupts        
+        regs.imsc.write(Interrupts::CTSM::SET + 
+                        Interrupts::RX::SET   + 
+                        Interrupts::TX::SET   + 
+                        Interrupts::RT::SET   + 
+                        Interrupts::FE::SET   + 
+                        Interrupts::PE::SET   + 
+                        Interrupts::BE::SET   + 
+                        Interrupts::OE::SET);
+    }
+
+
     pub fn handle_interrupt(&self) {
         let regs = unsafe { &*self.regs };
         // Get status bits
         #[allow(unused)]
         let flags: u32 = regs.fr.get();
+
         // Clear interrupts
         regs.icr.write(Interrupts::ALL_INTERRUPTS::SET);
+        regs.rsr_ecr.write(Errors::ALL_ERRORS::SET);
+
+        self.disable_interrupts();
+
+        if unsafe{udma::UDMA.transfer_complete(TX_DMA)} {
```

Comment:
I'd rather have the unsafe block outside of an `if-conditional`, such as
```
let transfer_complete = unsafe { ... };
if transfer_complete {
```
Otherwise the conditional becomes kind of messy